Sources/Controllers/Map/Widgets/OAMapWidgetRegistry.mm

```diff
@@ -36,22 +38,8 @@ - (instancetype) init
         _rightWidgetSet = [NSMutableOrderedSet orderedSet];
         _visibleElementsFromSettings = [NSMapTable strongToStrongObjectsMapTable];
         _settings = [OAAppSettings sharedManager];
-        
-        for (OAApplicationMode *ms in [OAApplicationMode values])
-        {
-            NSString *mpf = [_settings.mapInfoControls get:ms];
-            if ([mpf isEqualToString:SHOW_PREFIX])
-            {
-                [_visibleElementsFromSettings setObject:nil forKey:ms];
-            }
-            else
-            {
-                NSMutableSet<NSString *> *set = [NSMutableSet set];
-                [_visibleElementsFromSettings setObject:set forKey:ms];
-                NSArray<NSString *> *split = [mpf componentsSeparatedByString:SETTINGS_SEPARATOR];
-                [set addObjectsFromArray:split];
-            }
-        }
+        [self loadVisibleElementsFromSettings];
+        _widgetSettingsChangeDoneObserver = [[OAAutoObserverProxy alloc] initWith:self withHandler:@selector(onWidgetSettingsReset:withKey:) andObserve:[OsmAndApp instance].widgetSettingsChangingStartObservable];
```

Comment:
Names mismatch _widgetSettingsChangeDoneObserver and widgetSettingsChangingStartObservable. Looks like widgetSettingResetObservable should be used in both cases.